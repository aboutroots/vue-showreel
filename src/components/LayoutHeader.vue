<template>
  <div id="header" class="LayoutHeader">
    <Breadcrumbs />
    <LoginIndicator />
  </div>
</template>
<script lang="ts">
import Breadcrumbs from '@/components/LayoutBreadcrumbs.vue';
import LoginIndicator from '@/components/LoginIndicator.vue';

import Vue from 'vue';

export default Vue.extend({
  name: 'LayoutHeader',
  components: {
    Breadcrumbs,
    LoginIndicator,
  },
});
</script>

<style lang="scss" scoped>
@import '@/assets/styles/theme.scss';

#header {
  padding: 1rem;
  max-width: 100vw;
  width: 100%;
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap-reverse;

  // leave some space for hamburger button on mobile
  $ham-button-width: 60px;
  margin-left: $ham-button-width;
  width: calc(100vw - #{$ham-button-width});

  @include tabletUp {
    margin-left: $sidebar-width;
    width: calc(100vw - #{$sidebar-width} - 2rem);
  }
}
</style>
