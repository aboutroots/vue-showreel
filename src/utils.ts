export const sleep = async (ms: number): Promise<void> => {
  /**
   * Sleep for given number of miliseconds
   */
  // eslint-disable-next-line no-promise-executor-return
  await new Promise((r) => setTimeout(r, ms));
};
