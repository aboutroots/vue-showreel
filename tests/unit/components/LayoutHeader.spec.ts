import LayoutHeader from '@/components/LayoutHeader.vue';
import { shallowMount } from '../utils';

describe('LayoutHeader tests suite', () => {
  test('renders the component successfully', () => {
    const wrapper = shallowMount(LayoutHeader);

    const container = wrapper.find('.LayoutHeader');

    expect(container.exists()).toBeTruthy();
  });

  test('matches snapshot', () => {
    const wrapper = shallowMount(LayoutHeader);
    expect(wrapper.element).toMatchSnapshot();
  });
});
