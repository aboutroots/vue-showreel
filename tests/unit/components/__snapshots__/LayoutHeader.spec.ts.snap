// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutHeader tests suite matches snapshot 1`] = `
<div
  class="LayoutHeader"
  id="header"
>
  <breadcrumbs-stub />
   
  <loginindicator-stub />
</div>
`;
